assets/common/src/log_utils/logging_utils.py

```diff
@@ -0,0 +1,244 @@
+    AppInsightsPIIStrippingFormatter,
+)
+import platform
+import uuid
+import codecs
+import logging
+import sys
+
+
+class RunDetails:
+    """RunDetails."""
+
+    def __init__(self):
+        """Run details init."""
+        self._run = Run.get_context()
+
+    @property
+    def run_id(self):
+        """Run ID of the existing run."""
+        return self._run.id
+
+    @property
+    def parent_run_id(self):
+        """Parent RunID of the existing run."""
+        if "OfflineRun" in self.run_id:
+            return LoggerConfig.OFFLINE_RUN_MESSAGE
+        return self._run.parent.id
+
+    @property
+    def workspace(self):
+        """Return workspace."""
+        if "OfflineRun" in self.run_id:
+            return LoggerConfig.OFFLINE_RUN_MESSAGE
+        return self._run.experiment.workspace
+
+    @property
+    def workspace_name(self):
+        """Return workspace name."""
+        if "OfflineRun" in self.run_id:
+            return LoggerConfig.OFFLINE_RUN_MESSAGE
+        return self.workspace.name
+
+    @property
+    def experiment_id(self):
+        """Return experiment id."""
+        if "OfflineRun" in self.run_id:
+            return LoggerConfig.OFFLINE_RUN_MESSAGE
+        return self._run.experiment.id
+
+    @property
+    def subscription_id(self):
+        """Return subcription ID."""
+        if "OfflineRun" in self.run_id:
+            return LoggerConfig.OFFLINE_RUN_MESSAGE
+        return self.workspace.subscription_id
+
+    @property
+    def region(self):
+        """Return the region where the run is executing."""
+        if "OfflineRun" in self.run_id:
+            return LoggerConfig.OFFLINE_RUN_MESSAGE
+        return self.workspace.location
+
+    @property
+    def compute(self):
+        """Return compute target for the current run."""
+        if "OfflineRun" in self.run_id:
+            return LoggerConfig.OFFLINE_RUN_MESSAGE
+        details = self._run.get_details()
+        return details.get("target", "")
+
+    @property
+    def vm_size(self):
+        """Return compute VM size."""
+        if "OfflineRun" in self.run_id:
+            return LoggerConfig.OFFLINE_RUN_MESSAGE
+        compute_name = self.compute
+        if compute_name == "":
+            return "No compute found."
+        # TODO: Use V2 way of determining this.
+        cpu_cluster = ComputeTarget(workspace=self._run.experiment.workspace, name=compute_name)
+        return cpu_cluster.vm_size
+
+    @property
+    def root_attribute(self):
+        """Return root attribute of the run."""
+        if "OfflineRun" in self.run_id:
+            return LoggerConfig.OFFLINE_RUN_MESSAGE
+
+        cur_attribute = self._run.name
+        run = self._run.parent
+        # update current run's root_attribute to the root run.
+        while run is not None:
+            cur_attribute = run.name
+            run = run.parent
+        return cur_attribute
+
+    def __str__(self):
```

Comment:
Please take some of the logging changes from here as well: 
https://github.com/Azure/azureml-assets/pull/824/files

Reply:
Thanks , updated